beekeeper-api/src/test/java/com/expediagroup/beekeeper/api/BeekeeperControllerTest.java

```diff
@@ -0,0 +1,120 @@
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+import org.springframework.test.web.servlet.MockMvc;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import com.expediagroup.beekeeper.core.model.HousekeepingEntity;
+import com.expediagroup.beekeeper.core.model.HousekeepingMetadata;
+import com.expediagroup.beekeeper.core.repository.HousekeepingMetadataRepository;
+
+@ExtendWith(MockitoExtension.class)
+public class BeekeeperControllerTest {
+
+  @Autowired
+  private MockMvc mockMvc;
+  @Autowired
+  private ObjectMapper objectMapper;
+
+  @Mock
+  private Specification<HousekeepingMetadata> spec;
+  @Mock
+  private Pageable pageable;
+  @Mock
+  private HousekeepingEntityService housekeepingEntityService;
+
+  private HousekeepingMetadata table1;
+  private HousekeepingMetadata table2;
+
+  @BeforeEach
+  public void createTables(){
+
+    LocalDateTime CREATION_TIMESTAMP = LocalDateTime.now(ZoneId.of("UTC"));
+
+    table1 = new HousekeepingMetadata.Builder()
+        .path("s3://some/path/")
+        .databaseName("aRandomDatabase")
+        .tableName("aRandomTable")
+        .partitionName("event_date=2020-01-01/event_hour=0/event_type=A")
+        .housekeepingStatus(SCHEDULED)
+        .creationTimestamp(CREATION_TIMESTAMP)
+        .modifiedTimestamp(CREATION_TIMESTAMP)
+        .cleanupDelay(Duration.parse("P3D"))
+        .cleanupAttempts(0)
+        .lifecycleType(EXPIRED.toString())
+        .build();
+    table2 = new HousekeepingMetadata.Builder()
+        .path("s3://some/path2/")
+        .databaseName("aRandomDatabase2")
+        .tableName("aRandomTable2")
+        .partitionName("event_date=2020-01-012/event_hour=2/event_type=B")
+        .housekeepingStatus(DELETED)
+        .creationTimestamp(CREATION_TIMESTAMP)
+        .modifiedTimestamp(CREATION_TIMESTAMP)
+        .cleanupDelay(Duration.parse("P3D"))
+        .cleanupAttempts(0)
+        .lifecycleType(EXPIRED.toString())
+        .build();
+  }
+  
+  @Test
+  public void getAllTables() throws Exception {
+    List<HousekeepingMetadata> tablesList = new ArrayList<HousekeepingMetadata>();
+    tablesList.add(table1);
+    tablesList.add(table2);
+    Page<HousekeepingMetadata> tables = new PageImpl<>(tablesList);
+    System.out.println(tables.getContent());
+    when(housekeepingEntityService.returnAllTables(spec, pageable)).thenReturn(tables);
+
+    mockMvc
+        .perform(get(housekeepingEntityService.returnAllTables(spec, pageable).toString()))
+        .andDo(MockMvcResultHandlers.print())
+        .andExpect(status().isOk())
+        .andExpect(content().contentType(APPLICATION_JSON))
+        .andExpect(content().json(objectMapper.writeValueAsString(tables)));
```

Comment:
You should do verifications on the mock service to see that its only been called as expected